upcloud/resource_upcloud_objectstorage.go

```diff
@@ -0,0 +1,184 @@
+package upcloud
+
+import (
+	"context"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/request"
+	"github.com/UpCloudLtd/upcloud-go-api/upcloud/service"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func resourceUpCloudObjectStorage() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceObjectStorageCreate,
+		ReadContext:   resourceObjectStorageRead,
+		UpdateContext: resourceObjectStorageUpdate,
+		DeleteContext: resourceObjectStorageDelete,
+		Schema: map[string]*schema.Schema{
+			"size": {
+				Description:  "The size of the object storage bucket in gigabytes",
+				Type:         schema.TypeInt,
+				Required:     true,
+				ValidateFunc: validation.IntInSlice([]int{250, 500, 1000}),
+			},
+			"access_key": {
+				Description:  "The access key used to identify user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(4, 255),
+			},
+			"secret_key": {
+				Description:  "The secret key used to authenticate user",
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(8, 255),
+			},
+			"zone": {
+				Description: "The zone in which the object storage bucket will be created",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+			},
+			"name": {
+				Description: "The name of the object storage bucket to be created",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"description": {
+				Description: "The description of the object storage bucket to be created",
+				Optional:    true,
```

Comment:
hmm, could also set this to required, as cannot be updated later on to empty value (due to our go-api)

Reply:
Other option would be to default these to some sensible value. e.g. `Managed by terraform`